fre/app/regrid_xy/regrid_xy_rewrite.py

```diff
@@ -0,0 +1,257 @@
+  grid_spec = datadict["grid_spec"]
+  input_realm = datadict["component"]["inputRealm"]
+  
+  match input_realm:
+    case "atmos": mosaic_key = "atm_mosaic_file"
+    case "aerosol": mosaic_key = "atm_mosaic_file"
+    case "ocean": mosaic_key = "ocn_mosaic_file"
+    case "land": mosaic_key = "lnd_mosaic_file"    
+    
+  mosaic_file = str(xr.load_dataset(grid_spec)[mosaic_key].values.astype(str))
+    
+  if Path(input_dir+mosaic_file).exists():
+    fre_logger(f"Input mosaic file {input_dir}{mosaic_file} exists")
+  elif Path(mosaic_file).exists():
+    shutil.copy(mosaic_file, input_dir+mosaic_file)
+    fre_logger(f"Copying {mosaic_file} to input directory {input_dir}") 
+  else:
+    raise IOError(f"Input mosaic file {mosaic_file} could not be found")
+  
+  return mosaic_file
+      
+
+def get_grid_spec(grid_spec: str, datadict: dict) -> str:
+  
+  """
+  Gets the grid_spec file by first extracing the tar file,
+  where the tar file is the value of ppyaml["postprocess"]["settings"]["pp_grid_spec"]
+  in the pp yaml file.
+  
+  :grid_spec:  User-specified grid_spec filename.  
+               If not specified, grid_spec will be set to 'grid_spec.nc'
+  :type grid_spec: str
+  :datadict: dictionary containing relevant regrid parameters
+  :type datadict: dict
+  
+  :raises IOError:  Error if the grid_spec file cannot be found
+  """
+  
+  if grid_spec is None: grid_spec = "grid_spec.nc"
+    
+  #retrieve the grid_spec tar filename from ppyaml
+  pp_grid_spec_tar = datadict["pp_yaml"]['postprocess']['settings']['pp_grid_spec']
+    
+  #untar grid_spec tar file
+  if tarfile.is_tarfile(pp_grid_spec_tar):
+    with tarfile.open(pp_grid_spec_tar, 'r') as tar: tar.extractall() 
+        
+  if Path(grid_spec).exists():
+    fre_logger(f"Grid_spec file {grid_spec} found in {pp_grid_spec_tar}")
+    return grid_spec
+  else:
+    raise IOError(f"Grid_spec={grid_spec} could not be found in {pp_grid_spec_tar}")
+    
+
+def get_input_file(datadict: dict, history_file: str):
+  
+  """
+  Gets the 
+  """
+  
+  input_date = datadict["input_date"]
+  if input_date is None:
+    return history_file
+  else:
+    return f"{input_date}.{history_file}"
+
+
+def get_remap_file(datadict: dict):
+
+  remap_file = datadict["remap_file"]
+  input_dir = datadict["input_dir"]
+  input_mosaic = datadict["input_mosaic"]
+  nlon = datadict["output_nlon"]
+  nlat = datadict["output_nlat"]
+  interp_method = datadict["interp_method"]  
+    
+  remap_cache_file = f"{input_mosaic[:-3]}X{nlon}by{nlat}_{interp_method}.nc"
+  if remap_file is not None:
+    try:
+      shutil.copy(remap_file, input_dir+remap_cache_file)
+      fre_logger.info(f"Copied/Renamed {remap_file} to {remap_cache_file}")
+    except Exception as exc:
+      fre_logger.info(f"Cannot find specified remap_file {remap_file}\n"
+                      "Remap file {remap_cache_file} will be generated on-the-fly")
+
+  return remap_cache_file
+
+
+def get_scalar_fields(datadict: dict):
```

Comment:
There is now yaml specification for specifying lists of variables instead of all.

split-vars (run after regridding) will filter the variable list.

So to be most efficient, we shouldn't regrid variables we won't split out, but the pp files will be the same even without a change here.